keps/sig-architecture/20190412-conformance-behaviors.md

```diff
@@ -342,20 +350,97 @@ this is not a high risk.
 
 ## Design Details
 
-TBD.
+Delivery of this KEP shall be done in the following phases:
+
+### Phase 1
+
+In Phase 1, we will:
+* Implement the behavior formats and types described above. This will include
+  separate suites for tooling-generated behaviors and handcrafted behaviors.
+* Implement the directory structure described above to contain the behavior
+  lists, including how to tie tests back to behaviors.
+* `kubetestgen`, a tool which reads the OpenAPI schema and generates the list of
+  behaviors.
+* Migrate existing conformance tests to work with the new tooling. Existing
+  tooling around generation of conformance reports will not be changed in this
+  phase.
+
+#### Tying tests back to behaviors
+The proposal above mentions `tests.yaml` but does not describe a format for that
+file. The current conformance frameworks requests that during promotion of the
+test to conformance, the developer adds metadata, including the release name,
+the test name, and description. Tests are identified in the
+[conformance.txt](https://github.com/kubernetes/kubernetes/blob/master/test/conformance/testdata/conformance.txt)
+file by their Ginko description. Unfortunately, this does not produce unique
+test names, as it does not include all of the `Describe` calls from higher in
+the call tree (see this
+[slack discussion](https://kubernetes.slack.com/archives/C78F00H99/p1566324743171500)
+for more details).
+
+As part of this KEP, tests being promoted to conformance must add a unique
+identifier, `TestId`, in their conformance metadata. This will be used, along with
+the behavior IDs, to map which tests validate which behaviors in the
+`tests.yaml` file. The Go structures for `tests.yaml` are shown below.
+
+```go
+type BehaviorTestList struct {
+       Tests []BehaviorTest `json:"tests,omitempty"`
+}
+
+type BehaviorTest struct {
```

Comment:
I imagine a single test may cover several behaviors, certainly implicitly if not explicitly. This can reduce test run time even if it not ideal. In fact a single behavior may be covered by multiple tests also, though it's not necessary.